### ISSUE TITLE
`comparison-with-callable` warning on `property`-style descriptors

### ISSUE THREAD

**Issue by j-hil-nwm** [pylint-dev/pylint#9832]
### Bug description

```python
Big fan of pylint and everything you guys do!

Pylint doesn't seem to recognise custom `property`-style descriptors, and flags a `comparison-with-callable` warning.


# file1.py
from typing import Any, Callable


class MyProperty:

    def __init__(self, f: Callable[[Any], str]) -> None:
        self.f = f

    def __get__(self, owner: object | None, owner_cls: type) -> str:
        return self.f(owner) + f"In descriptor class:\n{owner=}\n{owner_cls=}\n"


class A:

    @MyProperty
    def x(self) -> str:
        return f"In owner class:\n{self=}\n"

    def __repr__(self) -> str:
        return "Instance of A"


EXPECTED = """\
In owner class:
self=Instance of A
In descriptor class:
owner=Instance of A
owner_cls=<class '__main__.A'>
"""

assert EXPECTED == A().x  # <- pylint doesn't like this


My actual use case is closer to the following and doesn't work either:

```py
# file2.py
from typing import Any, Callable, cast


class MyClassProperty:

    def __init__(self, f: Callable[[Any], str]) -> None:
        self.f = cast(classmethod, f).__func__

    def __get__(self, owner: object | None, owner_cls: type) -> str:
        return self.f(owner_cls) + f"In descriptor class:\n{owner=}\n{owner_cls=}\n"


class B:

    @MyClassProperty
    @classmethod
    def x(cls) -> str:
        return f"In owner class:\n{cls=}\n"

    def __repr__(self) -> str:
        return "Instance of B"


EXPECTED = """\
In owner class:
cls=<class '__main__.B'>
In descriptor class:
owner=None
owner_cls=<class '__main__.B'>
"""

assert EXPECTED == B.x  # <- doesn't like this either
```

Is this something pylint wants to support? Aside from disabling are there any work arounds? I'd be happy to just turn off this check in my code, my real issue is that downstream users will hit the same issue.
```


### Configuration

_No response_

### Command used

```shell
pylint --disable=all --enable=comparison-with-callable .\file1.py .\file2.py
```

### Pylint output

```python
************* Module file1
file1.py:31:7: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)
************* Module file2
file2.py:32:7: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)

------------------------------------------------------------------
Your code has been rated at 9.23/10 (previous run: 9.23/10, +0.00)
```

### Expected behavior

Ideally I'd expect `pylint` to look at the descriptor's `__get__` method to determine the type

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.11.7 (main, Jan 11 2024, 15:05:28) [MSC v.1927 64 bit (AMD64)]
```
